primitives/src/transaction.rs

```diff
@@ -748,6 +768,10 @@ pub struct TxIn {
 #[cfg(feature = "alloc")]
 impl TxIn {
     /// An empty transaction input with the previous output as for a coinbase transaction.
+    ///
+    /// This has a 0-byte scriptSig which is **invalid** per consensus rules
+    /// (coinbase scriptSig must be 2-100 bytes). This is kept for backwards compatibility
+    /// in PSBT workflows where the scriptSig is filled in later.
     pub const EMPTY_COINBASE: Self = Self {
```

Comment:
If this is only used in bodgy PSBT code then perhaps we should not have it in a 1.0.0 crate? Discussion not blocking merge of this PR though.